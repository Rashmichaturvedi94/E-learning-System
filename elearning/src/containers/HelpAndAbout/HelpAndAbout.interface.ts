export interface HelpAndAboutProps {
  // TODO: add required props below
  myProp?: boolean;
}
