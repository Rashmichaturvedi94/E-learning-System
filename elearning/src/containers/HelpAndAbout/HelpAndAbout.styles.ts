import { Text } from 'react-native';
import styled from 'styled-components/native';
import { Divider } from 'react-native-elements';

export const Title = styled(Text)`
  text-align: center;
  font-size: 30px;
  font-weight: bold;
  margin-top: 15px;
  margin-bottom: 20px;
`;

export const StyledDivider = styled(Divider)`
  background-color: black;
  height: 1px;
  margin-left: 16px;
  margin-right: 16px;
`;
