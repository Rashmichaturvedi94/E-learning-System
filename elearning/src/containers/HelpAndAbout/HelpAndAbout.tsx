import React from 'react';
import { CollapsibleView } from 'components/CollapsibleView';
import { ScrollView, View } from 'react-native';
import { Title, StyledDivider } from './HelpAndAbout.styles';

const qna = [
  {
    question: 'I forgot my password, what can I do?',
    answer:
      'click on the "Forgot your password?" Link on the login screen, enter e-mail address and link will be sent to you to set a new password.',
  },
  {
    question: 'How can I subscribe the course?',
    answer:
      'click on the course, click on the subscription button. You will be taken to the payment page where you can pay and subscribe the course.',
  },
  {
    question: 'Can I save the course?',
    answer: 'yes, you can save the course for future use.',
  },
  {
    question: 'Can I pause a course and continue at a later time?',
    answer:
      'The Course is divided into different modules. You can pause your learnign at any time and continue at a later time.',
  },
  {
    question: 'What technical equipment do I need for e-learning?',
    answer:
      'All you need is stable internet connection and device such as smartphone and tablet, It is also recommended to use headphones.',
  },
];

export const HelpAndAbout = () => (
  <View>
    <Title>FAQ</Title>
    <ScrollView>
      {qna.map((item, i) => (
        <View key={i.toString()}>
          <CollapsibleView question={item.question} answer={item.answer} />
          <StyledDivider />
        </View>
      ))}
    </ScrollView>
  </View>
);
