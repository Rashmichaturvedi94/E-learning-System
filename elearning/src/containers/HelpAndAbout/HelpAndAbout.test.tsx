import React from 'react';
import { render } from '@testing-library/react-native';
import { HelpAndAbout } from './HelpAndAbout';

describe('containers/HelpAndAbout', () => {
  it('should render', () => {
    const component = render(<HelpAndAbout />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
