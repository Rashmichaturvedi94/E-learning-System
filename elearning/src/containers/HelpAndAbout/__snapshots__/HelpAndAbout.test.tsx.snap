// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/HelpAndAbout should render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
          "marginBottom": 20,
          "marginTop": 15,
          "textAlign": "center",
        },
      ]
    }
  >
    FAQ
  </Text>
  <RCTScrollView>
    <View>
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                ]
              }
            >
              I forgot my password, what can I do?
            </Text>
          </View>
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              onLayout={[Function]}
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 16,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "marginTop": 0,
                    },
                  ]
                }
              >
                click on the "Forgot your password?" Link on the login screen, enter e-mail address and link will be sent to you to set a new password.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 1,
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
      </View>
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                ]
              }
            >
              How can I subscribe the course?
            </Text>
          </View>
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              onLayout={[Function]}
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 16,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "marginTop": 0,
                    },
                  ]
                }
              >
                click on the course, click on the subscription button. You will be taken to the payment page where you can pay and subscribe the course.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 1,
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
      </View>
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                ]
              }
            >
              Can I save the course?
            </Text>
          </View>
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              onLayout={[Function]}
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 16,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "marginTop": 0,
                    },
                  ]
                }
              >
                yes, you can save the course for future use.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 1,
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
      </View>
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                ]
              }
            >
              Can I pause a course and continue at a later time?
            </Text>
          </View>
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              onLayout={[Function]}
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 16,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "marginTop": 0,
                    },
                  ]
                }
              >
                The Course is divided into different modules. You can pause your learnign at any time and continue at a later time.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 1,
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
      </View>
      <View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "marginBottom": 16,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "marginTop": 16,
                  },
                ]
              }
            >
              What technical equipment do I need for e-learning?
            </Text>
          </View>
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
          >
            <View
              forwardedRef={[Function]}
              onLayout={[Function]}
              style={Object {}}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "gray",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "marginBottom": 16,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "marginTop": 0,
                    },
                  ]
                }
              >
                All you need is stable internet connection and device such as smartphone and tablet, It is also recommended to use headphones.
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "height": 1,
              "marginLeft": 16,
              "marginRight": 16,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
